crates/op-test-vectors/src/execution.rs

```diff
@@ -47,14 +51,43 @@ pub struct ExecutionReceipt {
     pub block_hash: B256,
     pub transaction_index: U256,
     #[serde(flatten)]
-    pub op_receipt: OpReceiptEnvelope,
+    pub inner: TypedReceipt<Log>,
+}
+
+impl From<Block> for ExecutionEnvironment {
+    fn from(block: Block) -> Self {
+        Self {
+            current_coinbase: block.header.beneficiary,
+            current_difficulty: block.header.difficulty,
+            current_gas_limit: U256::from(block.header.gas_limit),
+            previous_hash: block.header.parent_hash,
+            current_number: U256::from(block.header.number),
+            current_timestamp: U256::from(block.header.timestamp),
+            block_hashes: None,
+        }
+    }
+}
+
+impl From<TransactionReceipt<TypedReceipt<Log>>> for ExecutionReceipt {
+    fn from(receipt: TransactionReceipt<TypedReceipt<Log>>) -> ExecutionReceipt {
+        ExecutionReceipt {
+            transaction_hash: receipt.transaction_hash,
+            root: receipt.state_root.unwrap_or_default(),
+            contract_address: receipt.contract_address.unwrap_or_default(),
+            gas_used: U256::from(receipt.gas_used),
+            block_hash: receipt.block_hash.unwrap_or_default(),
+            transaction_index: U256::from(receipt.transaction_index.unwrap_or_default()),
+            inner: receipt.inner,
+        }
+    }
 }
```

Comment:
Maybe this should be a [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) so upstream, errors can be bubbled up.

Reply:
Then we can avoid needing to have all the `unwrap_or_default()`s and properly return errors if unset.